blocks/components/editable/index.js

```diff
@@ -101,14 +110,26 @@ export default class Editable extends wp.element.Component {
 		this.props.onFocus();
 	}
 
-	onChange() {
-		if ( ! this.editor.isDirty() ) {
+	isActive() {
+		return tinymce.EditorManager.activeEditor.id === this.editor.id;
+	}
+
+	onSelectionChange() {
+		// We must check this because selectionChange is a global event.
+		if ( ! this.isActive() ) {
 			return;
 		}
 
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.savedContent );
+		this.content = this.getContent();
+		this.selection = this.getSelection();
```

Comment:
Maybe we should do something like `this.props.onFocus( this.getSelection() )` instead of storing the selection locally. We'll get the selection back in the `focus` prop. And we'll be able to use it outside the `Editable` to perform the merge/split or any selection related action outside the editable itself.